lib/client/api.go

```diff
@@ -375,6 +375,11 @@ type Config struct {
 	// MockHeadlessLogin is used in tests for mocking the Headless login response.
 	MockHeadlessLogin SSHLoginFunc
 
+	// MockUserHomeDir mocks the user's home dir, useful for overriding Postgres
+	// and MySQL service conf file locations in parallel tests using temp dirs
+	// as the user's "home directory".
+	MockUserHomeDir string
+
 	// HomePath is where tsh stores profiles
 	HomePath string
```

Comment:
Can you make these comments clarify the difference between them and when each should be used if I'm a dev needing to write something to ~ or ~/.tsh/

Reply:
refactored to have an override path for each kind of db config file, which makes the difference clear

[4cf47ef](https://github.com/gravitational/teleport/pull/29220/commits/4cf47ef0faf998c4ffb9c707aec1990f21c5cfdf)